scripts/fixSpec.cjs

```diff
@@ -0,0 +1,667 @@
+          // Manual fixes
+          // TODO: should be fixed in okta-oas3
+          if (schemaKey === 'CreateIamRoleRequest' && propName === 'permissions') {
+            prop.items = {
+              '$ref': '#/components/schemas/Permission'
+            };
+          } else if (schemaKey === 'DevicePostureChecks' && propName === 'include') {
+            prop.items = {
+              type: 'object',
+              properties: {
+                variableName: { type: 'string' },
+                value: { type: 'string' }
+              }
+            };
+          } else if (schemaKey === 'Error409' && propName === 'errorCauses') {
+            prop.items = {
+              '$ref': '#/components/schemas/ErrorCause'
+            };
+          } else if (['PersonalAppsBlockList', 'RealmAssignment', 'RealmProfile'].includes(schemaKey) && propName === 'domains') {
+            prop.items = { type: 'string' };
+          } else if (schemaKey === 'RegistrationInlineHookResponse' && propName === 'commands') {
+            const registrationResponse = spec3.components.schemas['RegistrationResponse'];
+            if (registrationResponse) {
+              const items = registrationResponse.properties.commands?.items;
+              if (items) {
+                prop.items = _.clone(items);
+              }
+            }
+            if (!prop.items) {
+              prop.items = {
+                '$ref': '#/components/schemas/InlineHookResponseCommands'
+              };
+            }
+          } else {
+            console.warn(`! Can't resolve item type for array ${schemaKey}.${propName}`);
+          }
+
+          arrayPropsWithoutItems.push({ schemaKey, propName, fixed: !!prop.items });
+        }
+      }
+    }
+  }
+
+  return {
+    arrayPropsWithoutItems,
+  };
+}
+
+function loadSpec(yamlFile) {
+  const yamlStr = fs.readFileSync(yamlFile, { encoding: 'utf8' });
+  const yamlStrFixed = yamlStr
+    .replaceAll('../oauth/dist/oauth.yaml', 'oauth.yaml')
+    // to prevent warn in console
+    // [main] WARN  o.o.c.l.AbstractTypeScriptClientCodegen - Error (model name matches existing language type) cannot be used as a model name. Renamed to ModelError
+    .replaceAll("'#/components/schemas/Error'", "'#/components/schemas/ModelError'")
+    .replace(/^[ ]{4}Error:$/m, '    ModelError:');
+
+  const spec3 = yaml.load(yamlStrFixed);
+
+  // `src/client.js` exports `Client` class which is a base class for performing API requests
+  // But spec also have `Client` model which is not hoever yet used (it is used in `oauth.yaml` spec only)
+  // If it's not used we can safely delete `Client` model to prevent breaking changes
+  const doesSpecUseClientSchema = yamlStrFixed.indexOf('\'#/components/schemas/Client\'') !== -1;
+  if (!doesSpecUseClientSchema && isBackwardCompatibility) {
+    delete spec3.components.schemas['Client'];
+  }
+
+  return {
+    spec3, yamlStr
+  };
+}
+
+
+async function main() {
```

Comment:
Can you add some comments, explaining the reasons for this file and what it's doing